fastlane/Fastfile

```diff
@@ -1,117 +1,341 @@
+# Default build configuration by action
+build_configurations = {
+  test: "Development",
+  qa: "Alpha",
+  testflight: "Beta",
+  release: "Production"
+}
 
-# Use the iOS platform as default
-default_platform :ios
+# Default match types by build configuration
+match_types = {
+  test: "development",
+  qa: "adhoc",
+  testflight: "appstore",
+  release: "appstore"
+}
+
+# Default App ID names by build configuration
+app_names = {
+  test: "%s Dev",
+  qa: "%s Alpha",
+  testflight: "%s",
+  release: "%s"
+}
+
+release_bump_types = ["build", "patch", "minor", "major"]
+first_build = 0
+first_version = "0.0.0"
 
-# Define what to do for the iOS platform
 platform :ios do
+
   # Run this before doing anything else
   before_all do
 
-    # Uncomment this if you don't have the carthage frameworks(if you are deploying from your computer, you can leave this commented).
+    # Uncomment this if you don't have the carthage frameworks installed.
     # carthage(
     #   platform: "iOS",
     #   use_binaries: false,
     #   use_ssh: true
-    #   )
+    # )
 
   end
 
   # After all the steps have completed succesfully, run this.
   after_all do |lane|
 
-    # Remove all build artifacts created by fastlane to upload
+    desc "Remove all build artifacts created by fastlane to upload."
     clean_build_artifacts
 
   end
 
-  # If there was an error, run this
+  # If there was an error, run this.
   error do |lane, exception|
 
-    # Remove all build artifacts created by fastlane to upload
+    desc "Remove all build artifacts created by fastlane to upload."
     clean_build_artifacts
 
   end
 
-  # Build and publish the Alpha version to TestFlight
-  lane :release_testflight do
+  desc "Releases a new version to `TestFlight`. This lane uses the configuration mapped to `:testflight`."
+  desc ""
+  desc "It has basically 3 main responsabilities: build/version number managing, app building, and deploy."
+  desc ""
+  desc "- Reads the build/version number from `'CURRENT_BUILD_NUMBER'/'CURRENT_VERSION_NUMBER'` 
+xcode user defined build configuration for the configuration in use."
+  desc "- Increments and saves these new values in the xcode user defined build configuration."
+  desc "- Sets these incremented build and version values in the `Info.plist` to be used to build the app."
+  desc "- Builds the app using `gym` and `match` to get the signing identity. The provisioning profile in use is the one \
+selected in xcode for the configuration mapped by `:testflight`."
+  desc "- Uploads the generated `.dsym` file to `Rollbar`."
+  desc "- Discards the changes in `Info.plist`. Given this file is used for every configuration, these values are stored \
+in xcode user defined build configuration and just reflected in `Info.plist` during building."
+  desc "- Uploads the application to `TestFlight` using `pilot`."
+  desc ""
+  desc "Receives a parameter `bump_type` representing the type of deploy. It can be any of %s" % release_bump_types.to_s
```

Comment:
(Completely optional and not sure if it'd be correct, but I'd add a link to `http://semver.org/` for reference on what number to upgrade.)